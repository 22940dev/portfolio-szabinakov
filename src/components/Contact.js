import React, { useState } from 'react'
import axios from 'axios'

import '../styles/Contact.css'

const Contact = () => {

    const [state, setState] = useState({
        name: '',
        email: '',
        subject: '',
        message: ''
    })

    const [result, setResult] = useState(null)

    const sendEmail = (e) => {
        e.preventDefault()
        axios
        .post('/send', {...state})
        .then(response => {
            setResult(response.data)
            setState({name:'', email:'', subject:'', message:'' })
        })
        .catch(() => {
            setResult({ success: false, message: 'Something went wrong, Try again!3'})
        })
    }

    const onInputChange = (event) => {
        const {name, value} = event.target

        setState({
            ...state,
            [name]: value
        })

    }

    const resetForm = (event) => {
        setState({
            name:'',
            email:'',
            subject:'',
            message:''
        })
    }
    return(
        <div className='contactContainer'>
            {result && (
                <p>{result.message}</p>
            )}
            <form className='contactForm' onSubmit={sendEmail}>
                <div className='contactDetails'>
                    <label htmlFor='name'>Full Name
                    <input
                        data-testid='name'
                        type='text'
                        name='name'
                        value={state.name}
                        className='formInputName'
                        onChange={onInputChange}/>
                    </label>

                    <label htmlFor='email'>Email
                    <input 
                        data-testid='email'
                        type='text'
                        name='email'
                        value={state.email}
                        className='formInputEmail'
                        onChange={onInputChange}/>
                    </label>

                    <label htmlFor='subject'>Subject
                    <input 
                        type='text'
                        data-testid='subject'
                        name='subject'
                        value={state.subject}
                        className='formInputSubject'
                        onChange={onInputChange}/>
                    </label>

                    <label htmlFor='message'>Message
                    <textarea 
                        type='textarea'
                        data-testid='message'
                        name='message'
                        value={state.message}
                        className='textarea'
                        onChange={onInputChange}>
                    </textarea>
                    </label>
                </div>
                <div className='buttonContainer'>
                    <button 
                        type='submit'
                        data-testid='submitButton'
                        className='submitButtonForm'>
                            Send
                    </button>
                    <div>
                        <button 
                        type='reset'
                        data-testid='resetButton'
                        className='resetButtonForm'
                        onClick={resetForm}
                    >
                        Reset
                    </button>
                    </div>
                </div>
            </form>
        </div>
    )
}

export default Contact