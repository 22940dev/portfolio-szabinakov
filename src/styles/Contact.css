.contactContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 30px;
    font-weight: bold;
    color: white;
}

.contactForm {
    background-color: rgb(0, 35, 0);
    border: solid 4px  rgba(200, 180, 100);
    display: flex;
    flex-direction: column; 
    width: 300px;
    height: 420px;
    border-radius: 20px;
    box-shadow: 10px 1px 30px;
}

/* Details */


label:first-child {
    margin-top: 30px;
}

label {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin: 10px 0px 5px 0px;
    font-size: 13px;
    border: solid orange;
}
label > input {
    width: 220px;
    align-self: center;
}

label > .textarea {
    height: 120px;
    width: 220px;
    align-self: center;
}

label > input:hover,
label > .textarea:hover {
    border: solid rgb(223, 223, 2)
}


/* Buttons */

.buttonContainer {
    display: flex;
    justify-content: space-evenly;
    margin-top: 30px;
}
.submitButtonForm, .resetButtonForm {
    background-color: white;
    opacity: 90%;
    border: solid 1px rgba(200, 180, 100);
    border-radius: 3px;
    box-shadow: 1px 1px 1px;
    width: 50px;
    height: 20px;
}

.submitButtonForm:hover, .resetButtonForm:hover {
    background-color:  rgba(200, 180, 100);
}

@media screen and (min-width: 414px) {
    .contactForm {
        margin-top: 30px;
    }
}

@media screen and (min-width: 768px) {
    .contactForm {
        margin-top: 10px;
        width: 450px;
        height: 500px;
    }
    .contactDetails {
        width: 370px;
        align-self: center;
        margin-bottom: 15px;
    }

    .contactDetails:first-child {
        margin-top: 30px;
    }
    
    label {
        margin-bottom: 10px;
    }
    label > input {
        width: 90%;
    }
    label > .textarea {
        height: 150px;
        width: 90%;
    }
    
    
    
    /* Buttons */
    
    .submitButtonForm, .resetButtonForm {

        width: 65px;
        height: 27px;
    }
    
}