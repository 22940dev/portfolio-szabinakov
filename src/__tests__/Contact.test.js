import React from "react";
import { render } from "@testing-library/react";
import Contact from "../components/Contact";

describe("Contact", () => {
  it("matches the Snapshot", () => {
    const { asFragment } = render(<Contact />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders all the elements", () => {
    const { getByTestId } = render(<Contact />);

    expect(getByTestId("name")).toHaveClass("formInputName");
    expect(getByTestId("email")).toHaveClass("formInputEmail");
    expect(getByTestId("subject")).toHaveClass("formInputSubject");
    expect(getByTestId("message")).toHaveClass("textarea");
    expect(getByTestId("submitButton")).toHaveClass("submitButtonForm");
  });
});
