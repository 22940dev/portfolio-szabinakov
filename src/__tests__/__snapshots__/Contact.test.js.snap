// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact matches the Snapshot 1`] = `
<DocumentFragment>
  <div
    class="contactContainer"
  >
    <form
      class="contactForm"
    >
      <div
        class="contactDetails"
      >
        Send me a message!
        <label
          for="name"
        >
          <p
            class="labelText"
          >
            Full Name
          </p>
          <input
            class="formInputName"
            data-testid="name"
            name="name"
            type="text"
          />
        </label>
        <label
          for="email"
        >
          <p
            class="labelText"
          >
            Email
          </p>
          <input
            class="formInputEmail"
            data-testid="email"
            name="email"
            type="text"
          />
        </label>
        <label
          for="subject"
        >
          <p
            class="labelText"
          >
            Subject
          </p>
          <input
            class="formInputSubject"
            data-testid="subject"
            name="subject"
            type="text"
          />
        </label>
        <label
          for="message"
        >
          <p
            class="labelText"
          >
            Message
          </p>
          <textarea
            class="textarea"
            data-testid="message"
            name="message"
            type="textarea"
          />
        </label>
      </div>
      <button
        class="submitButtonForm"
        data-testid="submitButton"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</DocumentFragment>
`;
